packages/babel-helpers/src/helpers.js

```diff
@@ -506,10 +506,12 @@ helpers.setPrototypeOf = helper("7.0.0-beta.0")`
 
 helpers.isNativeReflectConstruct = helper("7.9.0")`
   export default function _isNativeReflectConstruct() {
-    if (typeof Reflect === "undefined" || !Reflect.construct) return false;
-
-    // core-js@3
-    if (Reflect.construct.sham) return false;
+    if (
+      typeof Reflect === "undefined" ||
+      !Reflect.construct ||
+      // core-js@3
+      Reflect.construct.sham
+    ) return false;
```

Comment:
Weak Nit: I actually liked the previous style better.

Reply:
Me too, but I noticed that merging them minifies better (terser doesn't merge the two if statements):

Reply:
It should, I made the PR myself 😉 : https://github.com/terser/terser/pull/555

Reply:
Not with this code :thinking:

```js
// write or paste code here

export function f() {
  if (!a) return false;
  if (!b) return false;
  try {
    foo();
    return true;
  } catch {
    return false;
  }
}
```

(tested at https://try.terser.org/)

Anyway, I'm happy to revert this change since it will be eventually handled by terser.

Reply:
Ahh, it's because it can't collapse the try-catch statement into an expression, so the if-statements exist only as a series of if-statements and not a nested ternary. Open an issue?